File: modules/@angular/router-deprecated/test/spies.ts
import {Location} from '@angular/common';
import {SpyObject, proxy} from '@angular/core/testing/testing_internal';
import {Router, RouterOutlet} from '@angular/router-deprecated';

export class SpyRouter extends SpyObject {
  constructor() { super(Router); }
}

export class SpyRouterOutlet extends SpyObject {
  constructor() { super(RouterOutlet); }
}

export class SpyLocation extends SpyObject {
  constructor() { super(Location); }
}

export class SpyPlatformLocation extends SpyObject {
  pathname: string = null;
  search: string = null;
  hash: string = null;
  constructor() { super(SpyPlatformLocation); }
}

File: modules/@angular/router-deprecated/test/spies.dart
library router.spies;

import 'package:angular2/platform/common.dart' show PlatformLocation, Location;
import 'package:angular2/router.dart';
import 'package:angular2/testing_internal.dart';

@proxy
class SpyLocation extends SpyObject implements Location {}

@proxy
class SpyRouter extends SpyObject implements Router {}

@proxy
class SpyRouterOutlet extends SpyObject implements RouterOutlet {}

class SpyPlatformLocation extends SpyObject implements PlatformLocation {
  String pathname = null;
  String search = null;
  String hash = null;
}
